### DISCUSSION TITLE
Using this firmware on non-Creality boards with a Creality (or any) screen.

### DISCUSSION THREAD

**Discussion by wagnerism** [mriscoc/Ender3V2S1#942] [Q&A]
I like the firmware and its features so much I would like to make it work on another board on an Ender 5 plus printer!  I have compiled firmware for another board with default touchscreen marlin.  I got that working so I'm looking to make it better, like this Professional Firmware.

Difficulty - I need five stepper motors for dual-Z, so the SKR mini won't work.

I have spent hours trying to figure this part out not knowing if it is possible.  I can figure out how to change the board type and customize the configuration for the physical printer - but am stuck on the connection to the Creality LCD screen or any graphical LCD screen.  I have been studying the pinouts and am having a difficult time setting the configuration and pins.

So let's take a random control board (I have a MKS Robin Nano V3.1) and look at how the pinouts are assigned to the EXP1, EXP2, or EXP3.  I have EXP1 with the +5V and ground pins, with the other eight pins connecting to pins on the SoC. 

Is it a simple matter of reassigning the marlin-named pins to the processor's pins in the EXP connectors?  If that's the case I won't even need to physically rewire anything.

The "RX" and "TX" on the display tells me it needs UART and not SPI, but I don't know if that's a configuration setting or even board capability.  Can any pin or only specific pins be used as LCD_SERIAL_PORT or do I need to find and connect some serial port header on the board?  There's a worthless wifi port on this board and it may have a serial port for configuration.

Or - am I making this bigger than it needs to be since I already have a default-graphic touchscreen working on this MKS board?  I could just adjust the processor, flash it, and see if it even loads.

Here's someone that connected a Creality screen to a BTT SKR board - but the pinout on the control board is "TX" and "RX"
https://www.smith3d.com/skr-mini-e3-v2-v3-on-ender-3-v2-ender-3-s1-lcd-dwin-knob-screen/
This requires rewiring the connection between the board and screen.

Does anybody have any guidance?

**Comment by lagnat** [comment]
BTT sells a cable to connect the E3V2 knob screen to the mini SKR.  I have tried it and it works fine.  I would also like to get dual Z stepper drivers going with this firmware so I'd love to keep this thread going.

**Comment by lagnat** [comment]
Just took a peek at the MKS Robin Nano V3.1.  Looks like both Z connector go to the same driver, so you'd be using the extra extruder driver for the other Z stepper?

**Reply by wagnerism** [reply]
That's exactly what I'm doing.

**Comment by wagnerism** [comment]
I tried switching the board setting and compiling quietly and quickly completes without any product.
